app/models/user_track.rb

```diff
@@ -55,9 +55,10 @@ def num_available_side_exercises
                 count
       +
       track.exercises.side.
-                where(unlocked_by: nil).
-                where.not(id: completed_side_exercise_ids).
-                count
+                      active.
+                      where(unlocked_by: nil).
```

Comment:
On a tangent, why don't we include side exercises that will be unlocked by a core exercise? Should we instead check which side exercises have been unlocked?

Reply:
"Bonus" exercises are side exercises that are unlocked by default at the start of the track. That's what this block of code covers. These exercises don't have solution records until someone actively clicks on them to start them, which is why we can't just check for all the ones that are unlocked. 

So it takes the exercises that you've unlocked in the bit above this, then adds ones that are unlocked by default here.

Reply:
Cool. Thanks :)